pallets/dispenser/src/lib.rs

```diff
@@ -0,0 +1,187 @@
+// Polimec Blockchain – https://www.polimec.org/
+// Copyright (C) Polimec 2022. All rights reserved.
+
+// The Polimec Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The Polimec Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@polimec.org
+
+#![cfg_attr(not(feature = "std"), no_std)]
+pub use pallet::*;
+
+pub use frame_support::traits::{
+	tokens::{currency::VestingSchedule, Balance},
+	Currency, ExistenceRequirement,
+};
+pub use polimec_common::credentials::{Did, EnsureOriginWithCredentials, InvestorType, UntrustedToken};
+pub use sp_runtime::traits::Convert;
+
+pub mod extensions;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod weights;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+pub type BalanceOf<T> = <CurrencyOf<T> as Currency<AccountIdOf<T>>>::Balance;
```

Comment:
We ware unfortunately stuck with Currency because of pallet vesting. We could probably implement both (currency for vesting and fungible for the transfer of tokens) but I rather keep it to 1 for simplicity. We can of course update once pallet_vesting is updated.